squidpy/pl/_spatial_utils.py

```diff
@@ -0,0 +1,960 @@
+        scalefactors = _get_unique_map(scalefactor_mapping)
+    except KeyError:
+        logg.debug("`scalefactors` set to `None`.")
+        scalefactors = None
+    if scalefactors is not None and img_res_key is not None:
+        if scale_factor is None:  # get intersection of scale_factor and match to img_res_key
+            scale_factor_key = [i for i in scalefactors if img_res_key in i]
+            if not len(scale_factor_key):
+                raise ValueError(f"No `scale_factor` found that could match `img_res_key`: {img_res_key}.")
+            _scale_factor_key = scale_factor_key[0]  # get first scale_factor
+            scale_factor = [
+                adata.uns[Key.uns.spatial][i][Key.uns.scalefactor_key][_scale_factor_key] for i in library_id
+            ]
+        else:  # handle case where scale_factor is float or list
+            scale_factor = _get_list(scale_factor, float, len(library_id), "scale_factor")
+
+        if size_key not in scalefactors and size is None:
+            raise ValueError(
+                f"Specified `size_key: {size_key}` does not exist and size is `None`, "
+                f"available keys are: `{scalefactors}`.\n Specify valid `size_key` or `size`."
+            )
+        if size is None:
+            size = 1.0
+        size = _get_list(size, float, len(library_id), "size")
+        if not (len(size) == len(library_id) == len(scale_factor)):
+            raise ValueError("Len of `size`, `library_id` and `scale_factor` do not match.")
+        size = [
+            adata.uns[Key.uns.spatial][i][Key.uns.scalefactor_key][size_key] * s * sf * 0.5
+            for i, s, sf in zip(library_id, size, scale_factor)
+        ]
+        return scale_factor, size
+    else:
+        scale_factor = 1.0 if scale_factor is None else scale_factor
+        scale_factor = _get_list(scale_factor, float, len(library_id), "scale_factor")
+
+        size = 120000 / adata.shape[0] if size is None else size
+        size = _get_list(size, Number, len(library_id), "size")
+        return scale_factor, size
+
+
+def _image_spatial_attrs(
+    adata: AnnData,
+    shape: _AvailShapes | None = None,
+    spatial_key: str = Key.obsm.spatial,
+    library_id: Sequence[str] | None = None,
+    library_key: str | None = None,
+    img: _SeqArray | bool = None,
+    img_res_key: str | None = Key.uns.image_res_key,
+    img_channel: int | None = None,
+    seg: _SeqArray | bool = None,
+    seg_key: str | None = None,
+    cell_id_key: str | None = None,
+    scale_factor: _SeqFloat = None,
+    size: _SeqFloat = None,
+    size_key: str | None = Key.uns.size_key,
+    img_cmap: Colormap | str | None = None,
+) -> SpatialParams:
+
+    library_id = _get_library_id(
+        adata=adata, shape=shape, spatial_key=spatial_key, library_id=library_id, library_key=library_key
+    )
+    if len(library_id) > 1 and library_key is None:
```

Comment:
Can this really happen? If yes, I'd say `_get_library_id` should check this (if it doesn't break anything).